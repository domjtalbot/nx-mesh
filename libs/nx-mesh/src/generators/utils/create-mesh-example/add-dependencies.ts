import type { Tree } from '@nrwl/devkit';

import type { CreateMeshExampleOptions } from './types';

import { addDependenciesToPackageJson } from '@nrwl/devkit';

import { versions } from '../../../utils/versions';
import examples from './examples';

export function addDependencies(
  host: Tree,
  options: Pick<CreateMeshExampleOptions, 'example'>
) {
  let exampleDeps = {};

  examples[options.example].dependencies.forEach((dep) => {
    exampleDeps = {
      ...exampleDeps,
      ...versions[dep],
    };
  });

  return addDependenciesToPackageJson(host, exampleDeps, {});
}

export default addDependencies;
