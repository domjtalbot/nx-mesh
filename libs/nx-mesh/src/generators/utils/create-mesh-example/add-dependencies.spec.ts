import type { Tree } from '@nrwl/devkit';
import type { CreateMeshExampleOptions, MeshDependencies } from './types';

import { getWorkspaceLayout, joinPathFragments } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

import { addDependencies } from './add-dependencies';

describe('addDependencies', () => {
  let tree: Tree;
  let projectDirectory: string;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();

    tree.write(
      'workspace.json',
      JSON.stringify({
        version: 2,
        projects: {},
      })
    );

    const workspace = getWorkspaceLayout(tree);

    projectDirectory = joinPathFragments(workspace.libsDir, 'test');
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it.each<CreateMeshExampleOptions['example']>([
    'country-info',
    'fake-api',
    'javascript-wiki',
    'movies',
    'rfam',
    'stackexchange',
    'star-wars-api',
    'trippin',
  ])(`should add %s dependencies to the package.json`, (example) => {
    addDependencies(tree, {
      example,
    });

    expect(tree.exists('package.json')).toBeTruthy();

    let dependenciesCount = 0;

    const packageJson = tree.read('package.json', 'utf-8');

    if (packageJson) {
      const parsedPackageJson = JSON.parse(packageJson);

      if (
        Object.prototype.hasOwnProperty.call(
          parsedPackageJson,
          'dependencies'
        ) &&
        typeof parsedPackageJson['dependencies'] === 'object'
      ) {
        dependenciesCount = Object.keys(
          parsedPackageJson['dependencies']
        ).length;
      }
    }

    expect(dependenciesCount).toBeGreaterThan(0);
    expect(packageJson).toMatchSnapshot();
  });
});
