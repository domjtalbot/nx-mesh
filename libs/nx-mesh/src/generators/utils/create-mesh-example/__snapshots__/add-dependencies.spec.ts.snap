// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addDependencies should add country-info dependencies to the package.json 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@graphql-mesh/cli\\": \\"0.79.3\\",
    \\"@graphql-mesh/runtime\\": \\"0.44.31\\",
    \\"@graphql-mesh/soap\\": \\"0.16.0\\",
    \\"@graphql-mesh/utils\\": \\"0.42.6\\",
    \\"graphql\\": \\"16.6.0\\"
  },
  \\"devDependencies\\": {}
}
"
`;

exports[`addDependencies should add fake-api dependencies to the package.json 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@graphql-mesh/cli\\": \\"0.79.3\\",
    \\"@graphql-mesh/json-schema\\": \\"0.35.33\\",
    \\"@graphql-mesh/plugin-mock\\": \\"0.0.37\\",
    \\"@graphql-mesh/runtime\\": \\"0.44.31\\",
    \\"@graphql-mesh/utils\\": \\"0.42.6\\",
    \\"graphql\\": \\"16.6.0\\"
  },
  \\"devDependencies\\": {}
}
"
`;

exports[`addDependencies should add javascript-wiki dependencies to the package.json 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@graphql-mesh/cli\\": \\"0.79.3\\",
    \\"@graphql-mesh/openapi\\": \\"0.33.34\\",
    \\"@graphql-mesh/runtime\\": \\"0.44.31\\",
    \\"@graphql-mesh/utils\\": \\"0.42.6\\",
    \\"graphql\\": \\"16.6.0\\"
  },
  \\"devDependencies\\": {}
}
"
`;

exports[`addDependencies should add movies dependencies to the package.json 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@graphql-mesh/cli\\": \\"0.79.3\\",
    \\"@graphql-mesh/neo4j\\": \\"0.20.34\\",
    \\"@graphql-mesh/runtime\\": \\"0.44.31\\",
    \\"@graphql-mesh/utils\\": \\"0.42.6\\",
    \\"graphql\\": \\"16.6.0\\"
  },
  \\"devDependencies\\": {}
}
"
`;

exports[`addDependencies should add rfam dependencies to the package.json 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@graphql-mesh/cli\\": \\"0.79.3\\",
    \\"@graphql-mesh/mysql\\": \\"0.17.30\\",
    \\"@graphql-mesh/runtime\\": \\"0.44.31\\",
    \\"@graphql-mesh/utils\\": \\"0.42.6\\",
    \\"graphql\\": \\"16.6.0\\"
  },
  \\"devDependencies\\": {}
}
"
`;

exports[`addDependencies should add stackexchange dependencies to the package.json 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@graphql-mesh/cli\\": \\"0.79.3\\",
    \\"@graphql-mesh/openapi\\": \\"0.33.34\\",
    \\"@graphql-mesh/runtime\\": \\"0.44.31\\",
    \\"@graphql-mesh/utils\\": \\"0.42.6\\",
    \\"graphql\\": \\"16.6.0\\"
  },
  \\"devDependencies\\": {}
}
"
`;

exports[`addDependencies should add star-wars-api dependencies to the package.json 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@graphql-mesh/cli\\": \\"0.79.3\\",
    \\"@graphql-mesh/graphql\\": \\"0.31.31\\",
    \\"@graphql-mesh/runtime\\": \\"0.44.31\\",
    \\"@graphql-mesh/utils\\": \\"0.42.6\\",
    \\"graphql\\": \\"16.6.0\\"
  },
  \\"devDependencies\\": {}
}
"
`;

exports[`addDependencies should add trippin dependencies to the package.json 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@graphql-mesh/cli\\": \\"0.79.3\\",
    \\"@graphql-mesh/odata\\": \\"0.20.31\\",
    \\"@graphql-mesh/runtime\\": \\"0.44.31\\",
    \\"@graphql-mesh/utils\\": \\"0.42.6\\",
    \\"graphql\\": \\"16.6.0\\"
  },
  \\"devDependencies\\": {}
}
"
`;
